packages/cli/src/commands/build.ts

```diff
@@ -608,6 +651,59 @@ async function doBuild(
   );
 }
 
+export async function determineFrameworkVersion(
+  frameworkRecord: UnboxPromise<ReturnType<typeof detectFrameworkRecord>>,
+  cwd: string,
+  output: Output
+) {
+  if (!frameworkRecord?.versionDependencies?.length) {
+    return;
+  }
+
+  for (let packageName of frameworkRecord.versionDependencies) {
+    const version = await lookupInstalledVersion(cwd, packageName, output);
+    if (version) {
+      return {
+        version,
+      };
+    }
+  }
+}
+
+interface NpmLSOutput {
+  dependencies?: {
+    [dependency: string]:
+      | {
+          version: string;
+        }
+      | undefined;
+  };
+}
+
+async function lookupInstalledVersion(
+  cwd: string,
+  packageName: string,
+  output: Output
+) {
+  try {
+    const args = `ls ${packageName} --depth=0 --json`.split(' ');
+    const { stdout } = await execa('npm', args, { cwd });
+
+    const lsOutput = readJSON<NpmLSOutput>(stdout);
+    if (!lsOutput || lsOutput instanceof CantParseJSON) {
+      const errorMessage = lsOutput ? `: ${lsOutput?.meta.innerError}` : '';
+      output.debug(
+        `Could not parse JSON output of \`npm ls\` for ${packageName}${errorMessage}`
+      );
+      return;
+    }
+
+    return lsOutput.dependencies?.[packageName]?.version;
+  } catch (error) {
+    output.debug(`Error during lookupInstalledVersion: ${error}`);
+  }
+}
```

Comment:
This currently only works when `npm ls` can retrieve the installed package, which works in npm, pnpm, and yarn 1 projects, at least.

I want to follow-up with some logic that will use the proper package manage to run `${packageManager} ls --depth=0` (and `--json` where supported), then parse out the result. This will be less brittle.

Reply:
Testing out [a `require` strategy`](https://github.com/vercel/vercel/pull/9009#discussion_r1045090544) instead.

Reply:
Updated in https://github.com/vercel/vercel/pull/9009#:~:text=Resolve%20conversation-,PR%20feedback,-fbf562e